File: Java/WoM_T3/src/main/java/sep3/service/interfaces/IMilkService.java
package sep3.service.interfaces;

import sep3.dto.MilkDtos;

public interface IMilkService {

    MilkDtos.MilkDto create(MilkDtos.CreateMilkDto dto);
    MilkDtos.MilkDto update(MilkDtos.UpdateMilkDto dto);
    void delete(long id);
    MilkDtos.MilkDto get(long id);
    MilkDtos.MilkListDto getAll();
    MilkDtos.MilkListDto getByContainer(MilkDtos.MilkByContainerQuery dto);

    // NEW
    MilkDtos.MilkDto approveForStorage(MilkDtos.ApproveMilkStorageDto dto);
}

File: C_Sharp/Server/WoM_WebApi/Services/Interfaces/IMilkService.cs
using ApiContracts;

namespace WoM_WebApi.Services.Interfaces;

// Service: milk records
public interface IMilkService
{
    Task<MilkDto> CreateAsync(CreateMilkDto dto);                 // create record
    Task<MilkDto> GetByIdAsync(long id);                          // get one
    Task<MilkListDto> GetAllAsync();                              // list all
    Task<MilkListDto> GetByContainerAsync(long containerId);      // filter by container
    Task<MilkDto> UpdateAsync(UpdateMilkDto dto);                 // update record
    Task ApproveStorageAsync(ApproveMilkStorageDto dto);          // approve/deny storage
    Task DeleteAsync(long id);                                    // delete record
}
